### ISSUE TITLE
Drop dependencies

### ISSUE THREAD

**Issue by J535D165** [thombashi/pathvalidate#4]
Thanks for your nice package. I would like to use and integrate it into another project. 

It would be nice if the package was a standalone package. There are two uncommon dependencies while only small parts of it are used. A package like ``typepy`` can be replaced easily. ``six`` is common, but you are using only one function of it. So why not include the following function from six.py (including the license): 
```
def add_metaclass(metaclass):
    """Class decorator for creating a class with a metaclass."""
    def wrapper(cls):
        orig_vars = cls.__dict__.copy()
        slots = orig_vars.get('__slots__')
        if slots is not None:
            if isinstance(slots, str):
                slots = [slots]
            for slots_var in slots:
                orig_vars.pop(slots_var)
        orig_vars.pop('__dict__', None)
        orig_vars.pop('__weakref__', None)
        return metaclass(cls.__name__, cls.__bases__, orig_vars)
    return wrapper
```

**Comment by thombashi** [comment]
Thank you for your suggestion.

>It would be nice if the package was a standalone package. 

`six` and `typepy` could replace easily, but other dependency package (`mbstrdecoder`) may need some effort.
I would like to keep packages independent, there are three reasons:

1. Dependency
    - `mbstrdecoder` also depend on `six` (using some of the functionalities)
2. Maintainability
    - `typepy` and `mbstrdecoder` is my packages, so, I would like to avoid duplicate code existance across packages
3. Enhancement
    - A standalone package will need an extra effort to enhancement (I intended to enhance `pathvalidate` by using some third party packages)

A standalone package is your mandatory requirement?

**Comment by J535D165** [comment]
It is not mandatory. But for a relative simple filename check, I need 3 packages. For me, that's a bit too much. Security plays also a big role, I need to know what's going on in the dependencies.

**Comment by thombashi** [comment]
Thank you for your answer.

If your concern is the number of dependent packages (lines of code that you need to be inspected?), below steps might be a workaround:

1. Replace `typepy` and `six` (as you described in the previous post) in `pathvalidate`
2. Include `six` to `mbstrdecorder` as a submodule
    - `mbstrdecorder` depends on `six` only 
3. Include `mbstrdecorder` to `pathvalidate` as a submodule

If I do these, `pathvalidate` will be independent from external packages.
Please let me know what you think. Is it still too much to you?

**Comment by thombashi** [comment]
@J535D165 
I've released `pathvalidate 0.15.0` which change to remove package dependencies.
I hope this will help you.

**Comment by J535D165** [comment]
Thanks a lot! Didn't expect this update, so it's great news. I think this will help many developers to integrate this valuable package into their projects.
